codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/message/MessageMarshallableGenerator.kt

```diff
@@ -175,14 +187,16 @@ class MessageMarshallableGenerator(
         write("headers.append(.init(name: \$S, value: .string(\$S)))", name, value)
     }
 
-    private fun renderPayloadSerialization(ctx: ProtocolGenerator.GenerationContext, writer: SwiftWriter, shape: Shape) {
+    private fun renderPayloadSerialization(ctx: ProtocolGenerator.GenerationContext, writer: SwiftWriter, memberShape: MemberShape) {
```

Comment:
Changed this to render serialization for the member shape so that other member info is available, not just the target shape.